# COGS 108 - Data Checkpoint

# Names

- Victoria Thai
- Hannah Yick
- Jane Dinh
- Natasha Supangkat
- Gabriel Ramiro

<a id='research_question'></a>
# Research Question

What trends can be found between a county’s designation of a superfund site and its socioeconomic/demographic trends? Does the demographic makeup of a county influence the amount of time between Superfund designation and the completion of the remediation process?

# Dataset(s)

We will combine our datasets using the common variable of geographic location (county). By analyzing both aspects - Superfund activity and demographic breakdown - holistically, we will be able to better visualize and understand trends and potentially a relationship between the two.

1) **Superfund/National Priorities List (NPL) Sites**
- Number of observations: 1327 (as of April 26, 2021)
- Features: Region, state, site name, site ID, EPA ID, address, city, zip, county, federal facility indicator (whether or not the site is a federal site), latitude, longitude, listing date.
 - We will be focusing on the site name, county, and listing date.
- Summary: This dataset provides a comprehensive overview of the main characteristics of current Superfund sites on the National Priorities List. Geographical information (county) can help us better understand the spread of the data by location, and will allow us to combine with demographic data to explore trends. Additionally, we will use the listing date to determine the source of funding (based on the date of policy changes) as well as to measure how long the site has been active.
- Source: https://semspub.epa.gov/work/HQ/201371.pdf 

2) **2019 American Community Survey 5 Year Estimate by County**
- Number of observations: 3220 
- Features: Total population, race, per capita income median household income, poverty status by race. 
- Summary: This dataset provides demographic and socioeconomic data on each city’s population. This will help us understand population trends based on the NPL sites. 
- Source: https://www.socialexplorer.com/explore-maps



# Setup

In [49]:
#Imports 
import pandas as pd
import numpy as np

#Graphing
import seaborn as sns
import matplotlib as mpl
import matplotlib.pyplot as plt
%matplotlib inline

#Statistics
import patsy
import statsmodels.api as sm
import scipy.stats as stats
from scipy.stats import ttest_ind, chisquare, normaltest

#Webscraping
import requests 
import bs4
from bs4 import BeautifulSoup

import warnings
warnings.filterwarnings('ignore')

# Read in the data and store it within a data frame
npl_df = npl_df = pd.read_csv('https://raw.githubusercontent.com/COGS108/group011_sp21/main/All%20current%20Final%20NPL%20Sites%20(FOIA%204).csv?token=AI2VKGXHTRP44IT6ZP5ASULASRZUM') 
gdrive_file_id = '1kNWY1Kc4RhQ9ttK7dTYtj0fhHE3BXOGh'
acs_df = pd.read_csv(f'https://docs.google.com/uc?id={gdrive_file_id}&export=download', encoding='ISO-8859-1')

# Data Cleaning

Describe your data cleaning steps here.

### National Priorities List (NPL) Superfund Data Cleaning

Let's take a look at the dataset we have for NPL sites:

In [50]:
npl_df

Region State  \
0          1    CT   
1          1    CT   
2          1    CT   
3          1    CT   
4          1    CT   
5          1    CT   
6          1    CT   
7          1    CT   
8          1    CT   
9          1    CT   
10         1    CT   
11         1    CT   
12         1    CT   
13         1    MA   
14         1    MA   
15         1    MA   
16         1    MA   
17         1    MA   
18         1    MA   
19         1    MA   
20         1    MA   
21         1    MA   
22         1    MA   
23         1    MA   
24         1    MA   
25         1    MA   
26         1    MA   
27         1    MA   
28         1    MA   
29         1    MA   
30         1    MA   
31         1    MA   
32         1    MA   
33         1    MA   
34         1    MA   
35         1    MA   
36         1    MA   
37         1    MA   
38         1    MA   
39         1    MA   
40         1    MA   
41         1    MA   
42         1    MA   
43         1    MA   
44         1    ME   
45         1    ME   
46         1    ME   
47         1    ME   
48         1    ME   
49         1    ME   
50         1    ME   
51         1    ME   
52         1    ME   
53         1    ME   
54         1    ME   
55         1    ME   
56         1    NH   
57         1    NH   
58         1    NH   
59         1    NH   
60         1    NH   
61         1    NH   
62         1    NH   
63         1    NH   
64         1    NH   
65         1    NH   
66         1    NH   
67         1    NH   
68         1    NH   
69         1    NH   
70         1    NH   
71         1    NH   
72         1    NH   
73         1    NH   
74         1    NH   
75         1    NH   
76         1    RI   
77         1    RI   
78         1    RI   
79         1    RI   
80         1    RI   
81         1    RI   
82         1    RI   
83         1    RI   
84         1    RI   
85         1    RI   
86         1    RI   
87         1    RI   
88         1    VT   
89         1    VT   
90         1    VT   
91         1    VT   
92         1    VT   
93         1    VT   
94         1    VT   
95         1    VT   
96         1    VT   
97         1    VT   
98         1    VT   
99         1    VT   
100        2    NJ   
101        2    NJ   
102        2    NJ   
103        2    NJ   
104        2    NJ   
105        2    NJ   
106        2    NJ   
107        2    NJ   
108        2    NJ   
109        2    NJ   
110        2    NJ   
111        2    NJ   
112        2    NJ   
113        2    NJ   
114        2    NJ   
115        2    NJ   
116        2    NJ   
117        2    NJ   
118        2    NJ   
119        2    NJ   
120        2    NJ   
121        2    NJ   
122        2    NJ   
123        2    NJ   
124        2    NJ   
125        2    NJ   
126        2    NJ   
127        2    NJ   
128        2    NJ   
129        2    NJ   
130        2    NJ   
131        2    NJ   
132        2    NJ   
133        2    NJ   
134        2    NJ   
135        2    NJ   
136        2    NJ   
137        2    NJ   
138        2    NJ   
139        2    NJ   
140        2    NJ   
141        2    NJ   
142        2    NJ   
143        2    NJ   
144        2    NJ   
145        2    NJ   
146        2    NJ   
147        2    NJ   
148        2    NJ   
149        2    NJ   
150        2    NJ   
151        2    NJ   
152        2    NJ   
153        2    NJ   
154        2    NJ   
155        2    NJ   
156        2    NJ   
157        2    NJ   
158        2    NJ   
159        2    NJ   
160        2    NJ   
161        2    NJ   
162        2    NJ   
163        2    NJ   
164        2    NJ   
165        2    NJ   
166        2    NJ   
167        2    NJ   
168        2    NJ   
169        2    NJ   
170        2    NJ   
171        2    NJ   
172        2    NJ   
173        2    NJ   
174        2    NJ   
175        2    NJ   
176        2    NJ   
177        2    NJ   
178        2    NJ   
179        2    NJ   
180        2    NJ   
18

We removed the following columns from the dataset because they are irrelevant to our research.  

In [51]:
npl_df = npl_df.drop(["Region", "Site ID", "EPA ID", "Address", "Zip", "FF Ind", "Latitude", "Longitude"], axis=1)
npl_df

State  \
0       CT   
1       CT   
2       CT   
3       CT   
4       CT   
5       CT   
6       CT   
7       CT   
8       CT   
9       CT   
10      CT   
11      CT   
12      CT   
13      MA   
14      MA   
15      MA   
16      MA   
17      MA   
18      MA   
19      MA   
20      MA   
21      MA   
22      MA   
23      MA   
24      MA   
25      MA   
26      MA   
27      MA   
28      MA   
29      MA   
30      MA   
31      MA   
32      MA   
33      MA   
34      MA   
35      MA   
36      MA   
37      MA   
38      MA   
39      MA   
40      MA   
41      MA   
42      MA   
43      MA   
44      ME   
45      ME   
46      ME   
47      ME   
48      ME   
49      ME   
50      ME   
51      ME   
52      ME   
53      ME   
54      ME   
55      ME   
56      NH   
57      NH   
58      NH   
59      NH   
60      NH   
61      NH   
62      NH   
63      NH   
64      NH   
65      NH   
66      NH   
67      NH   
68      NH   
69      NH   
70      NH   
71      NH   
72      NH   
73      NH   
74      NH   
75      NH   
76      RI   
77      RI   
78      RI   
79      RI   
80      RI   
81      RI   
82      RI   
83      RI   
84      RI   
85      RI   
86      RI   
87      RI   
88      VT   
89      VT   
90      VT   
91      VT   
92      VT   
93      VT   
94      VT   
95      VT   
96      VT   
97      VT   
98      VT   
99      VT   
100     NJ   
101     NJ   
102     NJ   
103     NJ   
104     NJ   
105     NJ   
106     NJ   
107     NJ   
108     NJ   
109     NJ   
110     NJ   
111     NJ   
112     NJ   
113     NJ   
114     NJ   
115     NJ   
116     NJ   
117     NJ   
118     NJ   
119     NJ   
120     NJ   
121     NJ   
122     NJ   
123     NJ   
124     NJ   
125     NJ   
126     NJ   
127     NJ   
128     NJ   
129     NJ   
130     NJ   
131     NJ   
132     NJ   
133     NJ   
134     NJ   
135     NJ   
136     NJ   
137     NJ   
138     NJ   
139     NJ   
140     NJ   
141     NJ   
142     NJ   
143     NJ   
144     NJ   
145     NJ   
146     NJ   
147     NJ   
148     NJ   
149     NJ   
150     NJ   
151     NJ   
152     NJ   
153     NJ   
154     NJ   
155     NJ   
156     NJ   
157     NJ   
158     NJ   
159     NJ   
160     NJ   
161     NJ   
162     NJ   
163     NJ   
164     NJ   
165     NJ   
166     NJ   
167     NJ   
168     NJ   
169     NJ   
170     NJ   
171     NJ   
172     NJ   
173     NJ   
174     NJ   
175     NJ   
176     NJ   
177     NJ   
178     NJ   
179     NJ   
180     NJ   
181     NJ   
182     NJ   
183     NJ   
184     NJ   
185     NJ   
186     NJ   
187     NJ   
188     NJ   
189     NJ   
190     NJ   
191     NJ   
192     NJ   
193     NJ   
194     NJ   
195     NJ   
196     NJ   
197     NJ   
198     NJ   
199     NJ   
200     NJ   
201     NJ   
202     NJ   
203     NJ   
204     NJ   
205     NJ   
206     NJ   
207     NJ   
208     NJ   
209     NJ   
210     NJ   
211     NJ   
212     NJ   
213     NJ   
214     NY   
215     NY   
216     NY   
217     NY   
218     NY   
219     NY   
220     NY   
221     NY   
222     NY   
223     NY   
224     NY   
225     NY   
226     NY   
227     NY   
228     NY   
229     NY   
230     NY   
231     NY   
232     NY   
233     NY   
234     NY   
235     NY   
236     NY   
237     NY   
238     NY   
239     NY   
240     NY   
241     NY   
242     NY   
243     NY   
244     NY   
245     NY   
246     NY   
247     NY   
248     NY   
249     NY   
250     NY   
251     NY   
252     NY   
253     NY   
254     NY   
255     NY   
256     NY   
257     NY   
258     NY   
259     NY   
260     NY   
261     NY   
262     NY   
263     NY   
264     NY   
265     NY   
266     NY   
267     NY   
268     NY   
269     NY   
270     NY   
271     NY   
272     NY   
273     NY   
274     NY   
275     NY   
276     NY   
277     NY   
278     NY   
279     NY   
280     NY   
281     NY   
282     NY   
283     NY   
284     NY   
2

***
### ACS Data Cleaning

Now let's take a look at our dataset for demographic data from the American Community Survey (ACS):

First, let's set the maximum rows displayed in the data frame to 5:

In [53]:
pd.options.display.max_rows = 5

In [54]:
acs_df

FIPS Geographic Identifier          Area Name  \
0     Geo_FIPS             Geo_GEOID           Geo_NAME   
1        01001          05000US01001     Autauga County   
...        ...                   ...                ...   
3219     72151          05000US72151  Yabucoa Municipio   
3220     72153          05000US72153    Yauco Municipio   

                     Qualifying Name State Postal Abbreviation Summary Level  \
0                          Geo_QName                Geo_STUSAB    Geo_SUMLEV   
1            Autauga County, Alabama                        al           050   
...                              ...                       ...           ...   
3219  Yabucoa Municipio, Puerto Rico                        pr           050   
3220    Yauco Municipio, Puerto Rico                        pr           050   

     Geographic Component File identification Logical Record Number      US  \
0             Geo_GEOCOMP          Geo_FILEID          Geo_LOGRECNO  Geo_US   
1                      00               ACSSF               0000013     NaN   
...                   ...                 ...                   ...     ...   
3219                   00               ACSSF               0000089     NaN   
3220                   00               ACSSF               0000090     NaN   

     Census Region Census Division State (Census Code) State (FIPS Code)  \
0       Geo_REGION    Geo_DIVISION         Geo_STATECE         Geo_STATE   
1              NaN             NaN                 NaN                01   
...            ...             ...                 ...               ...   
3219           NaN             NaN                 NaN                72   
3220           NaN             NaN                 NaN                72   

     County of current residence County Subdivision (FIPS) Place (FIPS Code)  \
0                     Geo_COUNTY                Geo_COUSUB         Geo_PLACE   
1                            001                       NaN               NaN   
...                          ...                       ...               ...   
3219                         151                       NaN               NaN   
3220                         153                       NaN               NaN   

     Place (State FIPS + Place FIPS) Census Tract Block Group  \
0                        Geo_PLACESE    Geo_TRACT  Geo_BLKGRP   
1                                NaN          NaN         NaN   
...                              ...          ...         ...   
3219                             NaN          NaN         NaN   
3220                             NaN          NaN         NaN   

     Consolidated City  \
0           Geo_CONCIT   
1                  NaN   
...                ...   
3219               NaN   
3220               NaN   

     American Indian Area/Alaska Native Area/Hawaiian Home Land (Census)  \
0                                                             Geo_AIANHH   
1                                                                    NaN   
...                                                                  ...   
3219                                                                 NaN   
3220                                                                 NaN   

     American Indian Area/Alaska Native Area/Hawaiian Home Land (FIPS)  \
0                                                         Geo_AIANHHFP   
1                                                                  NaN   
...                                                                ...   
3219                                                               NaN   
3220                                                               NaN   

     American Indian Trust Land/Hawaiian Home Land Indicator  \
0                                                Geo_AIHHTLI   
1                                                        NaN   
...                                                      ...   
3219                                                   

At a cursory look, the first row contains variable names we won't find useful. Hence we'll drop it then reset the index:

In [55]:
acs_df = acs_df.drop([0], axis=0)
acs_df.reset_index(drop=True, inplace=True)
acs_df

FIPS Geographic Identifier          Area Name  \
0     01001          05000US01001     Autauga County   
1     01003          05000US01003     Baldwin County   
...     ...                   ...                ...   
3218  72151          05000US72151  Yabucoa Municipio   
3219  72153          05000US72153    Yauco Municipio   

                     Qualifying Name State Postal Abbreviation Summary Level  \
0            Autauga County, Alabama                        al           050   
1            Baldwin County, Alabama                        al           050   
...                              ...                       ...           ...   
3218  Yabucoa Municipio, Puerto Rico                        pr           050   
3219    Yauco Municipio, Puerto Rico                        pr           050   

     Geographic Component File identification Logical Record Number   US  \
0                      00               ACSSF               0000013  NaN   
1                      00               ACSSF               0000014  NaN   
...                   ...                 ...                   ...  ...   
3218                   00               ACSSF               0000089  NaN   
3219                   00               ACSSF               0000090  NaN   

     Census Region Census Division State (Census Code) State (FIPS Code)  \
0              NaN             NaN                 NaN                01   
1              NaN             NaN                 NaN                01   
...            ...             ...                 ...               ...   
3218           NaN             NaN                 NaN                72   
3219           NaN             NaN                 NaN                72   

     County of current residence County Subdivision (FIPS) Place (FIPS Code)  \
0                            001                       NaN               NaN   
1                            003                       NaN               NaN   
...                          ...                       ...               ...   
3218                         151                       NaN               NaN   
3219                         153                       NaN               NaN   

     Place (State FIPS + Place FIPS) Census Tract Block Group  \
0                                NaN          NaN         NaN   
1                                NaN          NaN         NaN   
...                              ...          ...         ...   
3218                             NaN          NaN         NaN   
3219                             NaN          NaN         NaN   

     Consolidated City  \
0                  NaN   
1                  NaN   
...                ...   
3218               NaN   
3219               NaN   

     American Indian Area/Alaska Native Area/Hawaiian Home Land (Census)  \
0                                                                    NaN   
1                                                                    NaN   
...                                                                  ...   
3218                                                                 NaN   
3219                                                                 NaN   

     American Indian Area/Alaska Native Area/Hawaiian Home Land (FIPS)  \
0                                                                  NaN   
1                                                                  NaN   
...                                                                ...   
3218                                                               NaN   
3219                                                               NaN   

     American Indian Trust Land/Hawaiian Home Land Indicator  \
0                                                        NaN   
1                                                        NaN   
...                                                      ...   
3218                                                     NaN   
3219                    

It also looks as if there are several columns/rows for which there are no data for, i.e. NaN, so we'll drop the columns that contain NaN values across all rows and vice versa: 

In [56]:
acs_df.dropna(axis = 0, how = 'all', inplace = True)
acs_df.dropna(axis = 1, how = 'all', inplace = True)

In [57]:
acs_df

FIPS Geographic Identifier          Area Name  \
0     01001          05000US01001     Autauga County   
1     01003          05000US01003     Baldwin County   
...     ...                   ...                ...   
3218  72151          05000US72151  Yabucoa Municipio   
3219  72153          05000US72153    Yauco Municipio   

                     Qualifying Name State Postal Abbreviation Summary Level  \
0            Autauga County, Alabama                        al           050   
1            Baldwin County, Alabama                        al           050   
...                              ...                       ...           ...   
3218  Yabucoa Municipio, Puerto Rico                        pr           050   
3219    Yauco Municipio, Puerto Rico                        pr           050   

     Geographic Component File identification Logical Record Number  \
0                      00               ACSSF               0000013   
1                      00               ACSSF               0000014   
...                   ...                 ...                   ...   
3218                   00               ACSSF               0000089   
3219                   00               ACSSF               0000090   

     State (FIPS Code) County of current residence Total Population  \
0                   01                         001            55380   
1                   01                         003           212830   
...                ...                         ...              ...   
3218                72                         151            33499   
3219                72                         153            35428   

     Total Population.1 Population Density (Per Sq. Mile)       Area (Land)  \
0                 55380                          93.16273  594.443748774126   
1                212830                          133.8703  1589.82271732533   
...                 ...                               ...               ...   
3218              33499                          606.7053  55.2146125773556   
3219              35428                            523.22  67.7114797443077   

     Total Population: Total Population: White Alone  \
0                55380                         42527   
1               212830                        183471   
...                ...                           ...   
3218             33499                          6244   
3219             35428                         26607   

     Total Population: Black or African American Alone  \
0                                                10538   
1                                                19718   
...                                                ...   
3218                                             26418   
3219                                              1346   

     Total Population: American Indian and Alaska Native Alone  \
0                                                          140   
1                                                         1645   
...                                                        ...   
3218                                                         0   
3219                                                       159   

     Total Population: Asian Alone  \
0                              573   
1                             1969   
...                            ...   
3218                            11   
3219                             6   

     Total Population: Native Hawaiian and Other Pacific Islander Alone  \
0                                                                    26   
1                                                                     9   
...                                                                 ...   
3218                                                                  0   
3219                                                                  0   

     Total Population: Some Other Race Alone  \
0                                        378   
1

While no rows were dropped, for columns we went from 155 to 111. Let's take a look at the variables we have now: 

In [58]:
print(acs_df.columns.tolist())

['FIPS', 'Geographic Identifier', 'Area Name', 'Qualifying Name', 'State Postal Abbreviation', 'Summary Level', 'Geographic Component', 'File identification', 'Logical Record Number', 'State (FIPS Code)', 'County of current residence', 'Total Population', 'Total Population.1', 'Population Density (Per Sq. Mile)', 'Area (Land)', 'Total Population:', 'Total Population: White Alone', 'Total Population: Black or African American Alone', 'Total Population: American Indian and Alaska Native Alone', 'Total Population: Asian Alone', 'Total Population: Native Hawaiian and Other Pacific Islander Alone', 'Total Population: Some Other Race Alone', 'Total Population: Two or More Races', '% Total Population: White Alone', '% Total Population: Black or African American Alone', '% Total Population: American Indian and Alaska Native Alone', '% Total Population: Asian Alone', '% Total Population: Native Hawaiian and Other Pacific Islander Alone', '% Total Population: Some Other Race Alone', '% Total Pop

There seems to be several columns containing geographic information we won't find useful, so we'll drop those columns:

In [59]:
acs_df.drop(labels=['FIPS', 'Geographic Identifier', 'Summary Level', 'Geographic Component', 'File identification', 'Logical Record Number', 'State (FIPS Code)', 'County of current residence', 'Total Population.1', 'Population Density (Per Sq. Mile)', 'Area (Land)', 'Total Population:', 'Total Population: White Alone', 'Total Population: Black or African American Alone', 'Total Population: American Indian and Alaska Native Alone', 'Total Population: Asian Alone', 'Total Population: Native Hawaiian and Other Pacific Islander Alone', 'Total Population: Some Other Race Alone', 'Total Population: Two or More Races'], axis=1, inplace = True)

In [60]:
pd.options.display.max_columns = None
acs_df

Area Name                 Qualifying Name  \
0        Autauga County         Autauga County, Alabama   
1        Baldwin County         Baldwin County, Alabama   
...                 ...                             ...   
3218  Yabucoa Municipio  Yabucoa Municipio, Puerto Rico   
3219    Yauco Municipio    Yauco Municipio, Puerto Rico   

     State Postal Abbreviation Total Population  \
0                           al            55380   
1                           al           212830   
...                        ...              ...   
3218                        pr            33499   
3219                        pr            35428   

     % Total Population: White Alone  \
0                              76.79   
1                              86.21   
...                              ...   
3218                           18.64   
3219                            75.1   

     % Total Population: Black or African American Alone  \
0                                                  19.03   
1                                                   9.26   
...                                                  ...   
3218                                               78.86   
3219                                                 3.8   

     % Total Population: American Indian and Alaska Native Alone  \
0                                                           0.25   
1                                                           0.77   
...                                                          ...   
3218                                                           0   
3219                                                        0.45   

     % Total Population: Asian Alone  \
0                               1.03   
1                               0.93   
...                              ...   
3218                            0.03   
3219                            0.02   

     % Total Population: Native Hawaiian and Other Pacific Islander Alone  \
0                                                                    0.05   
1                                                                       0   
...                                                                   ...   
3218                                                                    0   
3219                                                                    0   

     % Total Population: Some Other Race Alone  \
0                                         0.68   
1                                         1.14   
...                                        ...   
3218                                      1.38   
3219                                     19.73   

     % Total Population: Two or More Races  \
0                                     2.16   
1                                     1.69   
...                                    ...   
3218                                  1.08   
3219                                   0.9   

     Median Household Income (In 2019 Inflation Adjusted Dollars)  \
0                                                           58731   
1                                                           58320   
...                                                           ...   
3218                                                        16295   
3219                                                        14743   

     Median Household Income (In 2019 Inflation Adjusted Dollars):  \
0                                                            58731   
1                                                            58320   
...                                                            ...   
3218                                                         16295   
3219                                                         14743   

     Median Household Income (In 2019 Inflation Adjusted Dollars): White Alone Householder  \
0                                                                                    66540   
1                                  

We're now down to 92 columns. Because our NPL dataset does not have for Puerto Rico, we're dropping the rows in this acs_data dataframe pertaining to Puerto Rico: 

In [61]:
acs_df.drop(index=acs_df[acs_df['State Postal Abbreviation'] == 'pr'].index, inplace=True)
acs_df

Area Name           Qualifying Name State Postal Abbreviation  \
0      Autauga County   Autauga County, Alabama                        al   
1      Baldwin County   Baldwin County, Alabama                        al   
...               ...                       ...                       ...   
3140  Washakie County  Washakie County, Wyoming                        wy   
3141    Weston County    Weston County, Wyoming                        wy   

     Total Population % Total Population: White Alone  \
0               55380                           76.79   
1              212830                           86.21   
...               ...                             ...   
3140             8027                           89.75   
3141             7049                           97.42   

     % Total Population: Black or African American Alone  \
0                                                  19.03   
1                                                   9.26   
...                                                  ...   
3140                                                0.04   
3141                                                0.23   

     % Total Population: American Indian and Alaska Native Alone  \
0                                                           0.25   
1                                                           0.77   
...                                                          ...   
3140                                                        0.95   
3141                                                        0.17   

     % Total Population: Asian Alone  \
0                               1.03   
1                               0.93   
...                              ...   
3140                               0   
3141                            0.77   

     % Total Population: Native Hawaiian and Other Pacific Islander Alone  \
0                                                                    0.05   
1                                                                       0   
...                                                                   ...   
3140                                                                    0   
3141                                                                 0.14   

     % Total Population: Some Other Race Alone  \
0                                         0.68   
1                                         1.14   
...                                        ...   
3140                                      5.48   
3141                                         0   

     % Total Population: Two or More Races  \
0                                     2.16   
1                                     1.69   
...                                    ...   
3140                                  3.79   
3141                                  1.28   

     Median Household Income (In 2019 Inflation Adjusted Dollars)  \
0                                                           58731   
1                                                           58320   
...                                                           ...   
3140                                                        54158   
3141                                                        57031   

     Median Household Income (In 2019 Inflation Adjusted Dollars):  \
0                                                            58731   
1                                                            58320   
...                                                            ...   
3140                                                         54158   
3141                                                         57031   

     Median Household Income (In 2019 Inflation Adjusted Dollars): White Alone Householder  \
0                                                                                    66540   
1                                                                                    61505   
...                                 

We ended up dropping 78 rows.

Furthermore, because the scope of our research question only considers demographics living below the poverty level, we are dropping columns pertaining to demographics living at or above the poverty level:

In [62]:
acs_df.drop(list(acs_df.filter(regex = 'At or Above Poverty Level')), axis = 1, inplace = True)
acs_df

Area Name           Qualifying Name State Postal Abbreviation  \
0      Autauga County   Autauga County, Alabama                        al   
1      Baldwin County   Baldwin County, Alabama                        al   
...               ...                       ...                       ...   
3140  Washakie County  Washakie County, Wyoming                        wy   
3141    Weston County    Weston County, Wyoming                        wy   

     Total Population % Total Population: White Alone  \
0               55380                           76.79   
1              212830                           86.21   
...               ...                             ...   
3140             8027                           89.75   
3141             7049                           97.42   

     % Total Population: Black or African American Alone  \
0                                                  19.03   
1                                                   9.26   
...                                                  ...   
3140                                                0.04   
3141                                                0.23   

     % Total Population: American Indian and Alaska Native Alone  \
0                                                           0.25   
1                                                           0.77   
...                                                          ...   
3140                                                        0.95   
3141                                                        0.17   

     % Total Population: Asian Alone  \
0                               1.03   
1                               0.93   
...                              ...   
3140                               0   
3141                            0.77   

     % Total Population: Native Hawaiian and Other Pacific Islander Alone  \
0                                                                    0.05   
1                                                                       0   
...                                                                   ...   
3140                                                                    0   
3141                                                                 0.14   

     % Total Population: Some Other Race Alone  \
0                                         0.68   
1                                         1.14   
...                                        ...   
3140                                      5.48   
3141                                         0   

     % Total Population: Two or More Races  \
0                                     2.16   
1                                     1.69   
...                                    ...   
3140                                  3.79   
3141                                  1.28   

     Median Household Income (In 2019 Inflation Adjusted Dollars)  \
0                                                           58731   
1                                                           58320   
...                                                           ...   
3140                                                        54158   
3141                                                        57031   

     Median Household Income (In 2019 Inflation Adjusted Dollars):  \
0                                                            58731   
1                                                            58320   
...                                                            ...   
3140                                                         54158   
3141                                                         57031   

     Median Household Income (In 2019 Inflation Adjusted Dollars): White Alone Householder  \
0                                                                                    66540   
1                                                                                    61505   
...                                 

We're now down to 68 columns. Let's take a closer look at the variable/column names alongside their indices: 

In [63]:
counter = 0
for col in acs_df.columns:
    print(counter, col)
    counter += 1

0 Area Name
1 Qualifying Name
2 State Postal Abbreviation
3 Total Population
4 % Total Population: White Alone
5 % Total Population: Black or African American Alone
6 % Total Population: American Indian and Alaska Native Alone
7 % Total Population: Asian Alone
8 % Total Population: Native Hawaiian and Other Pacific Islander Alone
9 % Total Population: Some Other Race Alone
10 % Total Population: Two or More Races
11 Median Household Income (In 2019 Inflation Adjusted Dollars)
12 Median Household Income (In 2019 Inflation Adjusted Dollars):
13 Median Household Income (In 2019 Inflation Adjusted Dollars): White Alone Householder
14 Median Household Income (In 2019 Inflation Adjusted Dollars): Black or African American Alone Householder
15 Median Household Income (In 2019 Inflation Adjusted Dollars): American Indian and Alaska Native Alone  Householder
16 Median Household Income (In 2019 Inflation Adjusted Dollars): Asian Alone
17 Median Household Income (In 2019 Inflation Adjusted Dollar

Our research question won't be considering per capita income on such a granular level, so we'll leave out the racial per capita income breakdown, i.e. columns 23-31. We also have a redundant column at index 11. So we will drop that as well. For columns 32-67, we have several redundant columns because raw numbers and percentages are included. We'll be only keeping the columns containing percentage data:

In [64]:
acs_df = acs_df.iloc[:, np.r_[0:11, 12:23, 34, 37, 40, 43, 46, 49, 52, 55, 58, 61, 64, 67]]

In [65]:
acs_df

Area Name           Qualifying Name State Postal Abbreviation  \
0      Autauga County   Autauga County, Alabama                        al   
1      Baldwin County   Baldwin County, Alabama                        al   
...               ...                       ...                       ...   
3140  Washakie County  Washakie County, Wyoming                        wy   
3141    Weston County    Weston County, Wyoming                        wy   

     Total Population % Total Population: White Alone  \
0               55380                           76.79   
1              212830                           86.21   
...               ...                             ...   
3140             8027                           89.75   
3141             7049                           97.42   

     % Total Population: Black or African American Alone  \
0                                                  19.03   
1                                                   9.26   
...                                                  ...   
3140                                                0.04   
3141                                                0.23   

     % Total Population: American Indian and Alaska Native Alone  \
0                                                           0.25   
1                                                           0.77   
...                                                          ...   
3140                                                        0.95   
3141                                                        0.17   

     % Total Population: Asian Alone  \
0                               1.03   
1                               0.93   
...                              ...   
3140                               0   
3141                            0.77   

     % Total Population: Native Hawaiian and Other Pacific Islander Alone  \
0                                                                    0.05   
1                                                                       0   
...                                                                   ...   
3140                                                                    0   
3141                                                                 0.14   

     % Total Population: Some Other Race Alone  \
0                                         0.68   
1                                         1.14   
...                                        ...   
3140                                      5.48   
3141                                         0   

     % Total Population: Two or More Races  \
0                                     2.16   
1                                     1.69   
...                                    ...   
3140                                  3.79   
3141                                  1.28   

     Median Household Income (In 2019 Inflation Adjusted Dollars):  \
0                                                            58731   
1                                                            58320   
...                                                            ...   
3140                                                         54158   
3141                                                         57031   

     Median Household Income (In 2019 Inflation Adjusted Dollars): White Alone Householder  \
0                                                                                    66540   
1                                                                                    61505   
...                                                                                    ...   
3140                                                                                 53480   
3141                                                                                 58081   

     Median Household Income (In 2019 Inflation Adjusted Dollars): Black or African American Alone Householder  \
0                                                     

We're now left with 34 variables/columns--more workable than the 155 we started off with.

Finally let's take a look at the breakdown of missing values from the resulting dataset:

In [66]:
pd.options.display.max_rows = None
pd.set_option('display.max_colwidth', None)
acs_missingness_df = (acs_df.isna()
            .sum()
            .to_frame('missing_count')
            .assign(missing_ratio = lambda x: x['missing_count']/len(acs_df))
            .loc[acs_df.isna().any()] )
print(acs_missingness_df)

                                                                                                                                        missing_count  \
Median Household Income (In 2019 Inflation Adjusted Dollars): White Alone Householder                                                               2   
Median Household Income (In 2019 Inflation Adjusted Dollars): Black or African American Alone Householder                                        1218   
Median Household Income (In 2019 Inflation Adjusted Dollars): American Indian and Alaska Native Alone  Householder                               1773   
Median Household Income (In 2019 Inflation Adjusted Dollars): Asian Alone                                                                        1772   
Median Household Income (In 2019 Inflation Adjusted Dollars): Native Hawaiian and Other Pacific Islander Alone  Householder                      2894   
Median Household Income (In 2019 Inflation Adjusted Dollars): Some Other Race Alon

There appears to be a lot of missingness for economic data when it comes to racial breakdowns, especially for more marginal groups such as Native Indian and Hawaiian groups. This may limit our analysis moving forward if we wanted to look beyond aggregate data and further analyze differences between racial groups. 
***